README.md

```diff
@@ -14,18 +16,139 @@ or
 
 Usage
 -----
-`capsula-hub run` if you install it globally
-`yarn/npm run` with npm scripts
+`capsula-hub run <token>`
 
+Right now, only the localhost mode is available.
+The `token` is the path to the served configuration file.
 
-Run it locally
---------------
-To install the dependencies: `yarn install`
+Configuration
+-------------
+The configuration should match this [API](https://github.com/capsulajs/capsulahub-core/blob/develop/packages/workspace/src/api/WorkspaceConfig.ts).
+
+Example: 
+```typescript
+export const capsulaHubConfig = {
+  name: 'my-config',
+  service: [
+    {
+      serviceName: 'myServiceA',
+      path: '@my-namespace/my-service-a',
+      definition: {
+        serviceName: 'myServiceA',
+        methods: {
+          myMethod1: { asyncModel: 'RequestResponse' },
+          myMethod2$: { asyncModel: 'RequestStream' },
+        }
+      },
+      config: {}
+    },
+    {
+      serviceName: 'myServiceB',
+      path: 'https://my-cdn/my-service-b',
+      definition: {
+        serviceName: 'myServiceB',
+        methods: {
+          myMethod1: { asyncModel: 'RequestResponse' },
+          myMethod2: { asyncModel: 'RequestResponse' },
+        }
+      },
+      config: {}
+    },
+  ],
+  components: {
+    layouts: {
+      grid: {
+        componentName: 'web-grid',
+        nodeId: 'root',
+        path: 'http://my-cdn/components/Grid.js',
+        config: { title: 'Base Grid' },
+       },
+    },
+    items: {
+      ['request-form']: {
+        componentName: 'web-request-form',
+        nodeId: 'request-form',
+        path: 'http://my-cdn/components/RequestForm.js',
+        config: { title: 'Base Request Form' },
+      },
+    }
+  }
+};
+```
+
+Develop your extension
+----------------------
+An extension is a service that is able to register himself and can be reached by the configuration
+as detailed in the previous section.
+
+This service can be reached when deployed to npm registry, any CDN or even served locally.
+
+The two variables `WORKSPACE` and `SERVICE_CONFIG` are passed through the service `Workspace` which 
+is a core service.
+
+`WORKSPACE` contains the methods that allows the service to register itself.
+`SERVICE_CONFIG` contains the service specific configuration previously loaded with the `token`.
+
+```typescript
+const bootstrap = (WORKSPACE: any, SERVICE_CONFIG: any) => {
```

Comment:
It will be good also to have an example of extension that just run some code (flows), this kind of extension is very simple, it just use `WORKSPACE` to run the flows and don't need to even register itself.